core/node/fee_model/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -95,25 +93,29 @@ impl GasAdjuster {
         let last_processed_block = self.base_fee_statistics.last_processed_block();
 
         if current_block > last_processed_block {
-            let (base_fee_history, blob_base_fee_history) = Self::get_base_fees_history(
-                self.eth_client.as_ref(),
-                (last_processed_block + 1)..=current_block,
-            )
-            .await?;
+            let n_blocks = current_block - last_processed_block;
+            let base_fees = self
+                .eth_client
+                .base_fee_history(current_block, n_blocks)
+                .await?;
 
             // We shouldn't rely on L1 provider to return consistent results, so we check that we have at least one new sample.
-            if let Some(current_base_fee_per_gas) = base_fee_history.last() {
+            if let Some(current_base_fee_per_gas) = base_fees.last().map(|fee| fee.base_fee_per_gas)
+            {
                 METRICS
                     .current_base_fee_per_gas
-                    .set(*current_base_fee_per_gas);
+                    .set(current_base_fee_per_gas);
             }
-            self.base_fee_statistics.add_samples(&base_fee_history);
+            self.base_fee_statistics
+                .add_samples(base_fees.iter().map(|fee| fee.base_fee_per_gas));
 
-            if let Some(current_blob_base_fee) = blob_base_fee_history.last() {
+            if let Some(current_blob_base_fee) =
+                base_fees.last().map(|fee| fee.base_fee_per_blob_gas)
+            {
                 // Blob base fee overflows `u64` only in very extreme cases.
                 // It doesn't worth to observe exact value with metric because anyway values that can be used
                 // are capped by `self.config.max_blob_base_fee()` of `u64` type.
-                if current_blob_base_fee > &U256::from(u64::MAX) {
+                if current_blob_base_fee > U256::from(u64::MAX) {
                     tracing::error!("Failed to report current_blob_base_fee = {current_blob_base_fee}, it exceeds u64::MAX");
```

Comment:
I mean, it's possible to redefine the metric as `f64` if we really want to. Not sure whether it makes sense from the semantic standpoint though.

Reply:
Up to @RomanBrodetski 
I personally think that if the blob price will reach `u64::MAX`, we'll know it anyway 😅